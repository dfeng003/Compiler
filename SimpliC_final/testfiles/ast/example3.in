int gcd2(int a, int b) {
if (b == 0) {
return a;
}
return gcd2(b, a );
}