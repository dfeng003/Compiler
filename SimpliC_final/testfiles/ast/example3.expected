Program
  List
    Func_del
      IdFun ID="gcd2"
      List
        Parafactor
          IdVar ID="a"
        Parafactor
          IdVar ID="b"
      List
        IfStmt
          Eq
            IdUse ID="b"
            Numeral NUMERAL="0"
          Block
            List
              Return
                IdUse ID="a"
          Opt
        Return
          Func_call
            IdCall ID="gcd2"
            List
              IdUse ID="b"
              IdUse ID="a"
